src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/util/OcrDataDeserializer.java

```diff
@@ -1,19 +1,35 @@
-package uk.gov.hmcts.reform.bulkscanprocessor.services;
+package uk.gov.hmcts.reform.bulkscanprocessor.util;
 
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
 import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.OcrDataParseException;
 import uk.gov.hmcts.reform.bulkscanprocessor.model.ocr.OcrData;
 
+import java.io.IOException;
 import java.util.Base64;
 import java.util.Map;
 
 import static java.util.stream.Collectors.toMap;
 
-public class OcrDataParser {
+public class OcrDataDeserializer extends StdDeserializer<Map<String, String>> {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
-    public Map<String, String> parseOcrData(String base64EncodedOcrData) {
+    public OcrDataDeserializer() {
+        super(Map.class);
+    }
+
+    @Override
+    public Map<String, String> deserialize(
+        JsonParser jsonParser,
+        DeserializationContext deserializationContext
+    ) throws IOException {
+        return parseOcrData(jsonParser.getText());
```

Comment:
Pretty pointless to have a method which just calls another method and there's no other use of that child method anywhere else. How about we get rid of try/catch block in the private method and wrap here by using built-in Jackson process of deserialising incoming string? This will use the library effectively.

```java
try {
   return parseOcrData(jsonParser.getText());
} catch (<list of exceptions> exception) {
    throw deserializationContext.instantiationException(_valueClass, exception);
}
```

Not entirely sure deserialization context actually needs throw part here

Reply:
It's actually two statements happening here, not just a method call. I'm ok with moving try-catch to this method. Will do. Not sure about instantiationException, as it seems to have a more specific purpose (problem with constructor).

Reply:
Was just a suggestion to use context 😉. I think ideally we all should just try use more features from libraries we already incorporate. Of course, if it makes sense to do so. Deserialisation seemed like a good start. It is construct - it is trying to construct a map for ocr data field. In case there can be more concrete problem identified - different method used.

Like I said, was just a suggestion